src/System.CommandLine.Subsystems/HelpSubsystem.cs

```diff
@@ -24,8 +24,20 @@ public class HelpSubsystem(IAnnotationProvider? annotationProvider = null)
         Arity = ArgumentArity.Zero
     };
 
-    protected internal override void Initialize(InitializationContext context) 
-        => context.Configuration.RootCommand.Add(HelpOption);
+    protected internal override void Initialize(InitializationContext context)
+    {
+        AddOptionRecursively(context.Configuration.RootCommand, HelpOption);
+
+        static void AddOptionRecursively(CliCommand command, CliOption option)
```

Comment:
I imagine this method should belong to CliCommand or some extensions method, but I didn't want to change too much in this PR without consulting you first.